acrobat/blocks/dc-converter-widget/dc-converter-widget.js

```diff
@@ -206,20 +221,20 @@ export default async function init(element) {
   widgetContainer.className = `fsw wapper-${VERB}`;
   widget.appendChild(widgetContainer);
 
-  const isReturningUser = window.localStorage.getItem('pdfnow.auth');
   const isRedirection = /redirect_(?:conversion|files)=true/.test(window.location.search);
-  const preRenderDropZone = !isReturningUser && !isRedirection;
-
-  const verbIncludeList = ['compress-pdf', 'fillsign', 'sendforsignature', 'add-comment',
-    'delete-pages', 'reorder-pages', 'split-pdf', 'insert-pdf', 'extract-pages', 'crop-pages', 'number-pages'];
+  const { cookie } = document;
+  const limitCookie = exhLimitCookieMap[VERB] || exhLimitCookieMap[VERB.match(/^pdf-to|to-pdf$/)?.[0]];
+  const cookiePrefix = appEnvCookieMap[ENV] || '';
+  const isLimitExhausted = cookie.includes(`${cookiePrefix}${limitCookie}`);
+  const preRenderDropZone = !isLimitExhausted && !isRedirection;
 
   const INLINE_SNIPPET = widget.querySelector(':scope > section#edge-snippet');
   if (INLINE_SNIPPET) {
     widgetContainer.dataset.rendered = 'true';
     widgetContainer.appendChild(...INLINE_SNIPPET.childNodes);
     widget.removeChild(INLINE_SNIPPET);
     performance.mark('milo-move-snippet');
```

Comment:
In the case where the edge worker inlined the snippet, we will want to display it (copy it over) only if the limits have not been exhausted.   I the limits have been exhausted, just remove the snippet from the DOM.

```suggestion
  if (INLINE_SNIPPET) {
    if (!isLimitExhausted) {
      widgetContainer.dataset.rendered = 'true';
      widgetContainer.appendChild(...INLINE_SNIPPET.childNodes);
      performance.mark('milo-move-snippet');
    }
    widget.removeChild(INLINE_SNIPPET);
```

Reply:
Updated